### ISSUE TITLE
Invalid wasm (trying to access parent frame stack values) with latest nightly (wasm32-unknown-unknown)

### ISSUE THREAD

**Issue by DarrenTsung** [rust-lang/rust#50869]
Hi, I'm running into an issue I believe introduced in the nightly compiler sometime after `nightly (8ff4b4206 2018-05-08)`. I filed another issue in `wasm-bindgen` earlier: https://github.com/rustwasm/wasm-bindgen/issues/199, but believe that it is a compiler change that caused this to occur.

I've created a test-case repository here: https://github.com/DarrenTsung/wasm-validation-64.

On my current nightly, the wasm code produced fails validation with [wasmi](https://github.com/paritytech/wasmi) with the error: `Trying to access parent frame stack values.`, but this does not occur with nightly-2018-05-09. 

I've included the rustc version that I ran against in the test-case repository, but please let me know if any other information would be useful!

**Comment by alexcrichton** [comment]
Thanks for the report here @DarrenTsung, I've been able to reproduce with these instructions!

It looks like somewhere along the way LLVM is producing an invalid object file for wasm. Using `wasm-validate` I get:

```
error: type mismatch in call, expected [i32, i32] but got [i32]
```

Sure enough using `llc` to reassemble one of the intermediate bytecode files I'm also able to generate the same error.

I've opened an [upstream bug](https://bugs.llvm.org/show_bug.cgi?id=37546) to track this for LLVM
